src/SRSMain.cpp

```diff
@@ -28,128 +29,91 @@ SRSMain::Reprocess(){
-              eventBuilder = new SRSEventBuilder(nEvent,_conf->GetMaxClusterSize(), _conf->GetMinClusterSize(), _conf->GetZeroSupCut(), _conf->GetRunType(), 
-						 std::string(_conf->GetClusterPositionCorrectionFlag()).find("applyCorrections")!=std::string(_conf->GetClusterPositionCorrectionFlag()).npos);
-            }
-	    if(debug_s)	std::cout << " >>>> BAD! Merging two fragment in  "<<nEvent<< " tFEC "<<iFEC<<" # word "<<previous.size()<<std::endl;
-	    //decode previous                                                                                                                                                                            
-            for(unsigned int i=0;i<previous.size();i++){
-              event[i]=previous[i];
-            }
-	    if (debug_s) std::cout <<" +++ reprocess: event # "<<nEvent<<" FEC "<<tFEC<<std::endl;
-            SRSFECDecoder dec(eventBuilder);
-            dec.decodeFEC(previous.size(),event);
-            previous.clear();
-          }	  
-        }else{
-	  if(debug_s) std::cout <<" >>>> SKIP the decoding waiting next fragment"<<std::endl;
-          for (unsigned int ir=0; ir < nw-start_eventRecord;ir++){
-            previous.push_back(buffer[start_eventRecord+ir]);
-          }
-          prenevt=nEvent;
+        nadc++;
+        int adcId = j&0xff;//(( j >>8 ) & 0xffffffff);                                                                                  
+	if (debug_s)std::cout <<"SRSMain::  +++ reprocess: "<<std::hex<<" ADC Id = 0x"<<adcId<<std::dec<<std::endl;
+	//	goaway = (nadc == 35);
+      } else if ((j&0xffffffff) == 0xfafafafa){
+        firstADCinFEC = true;
+        fillFECdata = false;
+        iw++;
+        buffer[iw]=j;
+        if (iFEC == 1) {
+	  eventBuilder = new SRSEventBuilder(nEvent,_conf->GetMaxClusterSize(), _conf->GetMinClusterSize(), _conf->GetZeroSupCut(),\
+					     _conf->GetRunType(),
+					     std::string(_conf->GetClusterPositionCorrectionFlag()).find("applyCorrections")!=std::\
+					     string(_conf->GetClusterPositionCorrectionFlag()).npos);
+
+          nEvent++;
+	  if(debug_s)std::cout <<"New Event "<<nEvent<<std::endl;
         }
-        if (tFEC == nFEC){
-          // The event is now complete and we can form the cluster
+        if (iFEC == nFEC){
+	  SRSFECDecoder dec(eventBuilder);
+	  dec.decodeFEC(iw,buffer);
+	  if (nEvent%1000==0) std::cout<<" SRSMain:: +++ reprocess: Event # "<<nEvent<<std::endl;
 	  if (debug_s) std::cout <<" +++ reprocess: compute cluster"<<std::endl;
           eventBuilder->ComputeClustersInDetectorPlane();
-          // Fill the trees
+          // Fill the trees                                                                                                             
           _root->FillRootFile(eventBuilder);
           eventBuilder=0;
-	  
-        }
-        if (iFEC == 2) {
-          newevent = true;
-          //  delete eventBuilder;
-          iFEC = 0;
+
+        }else if (iFEC > 3){
```

Comment:
Indeed this is working for up 3 FECs. Checked with many files taken on QC5.
It will not work for 4 FECs. The reason for "else if (iFEC > 3)" instead of "else if (iFEC > nFEC)" is because of nFEC = 1 with FECid = 3. 
I propose to put this version in production for a while, in the mean time I will code some more robust check on the consistency between the config file and the actual data content